ksql-functional-tests/src/test/resources/rest-query-validation-tests/insert-values.json

```
@@ -66,6 +66,138 @@
         {"topic": "test_topic", "key": "10", "value": {"ID": null}}
       ]
     },
+    {
+      "name": "rowkey should be set when stream has int key and only id specified in insert",
+      "statements": [
+        "CREATE STREAM TEST (ID INT) WITH (kafka_topic='test_topic', value_format='JSON', key='ID');",
+        "INSERT INTO TEST (ID) VALUES (10);"
+      ],
+      "inputs": [
+      ],
+      "outputs": [
+        {"topic": "test_topic", "key": "10", "value": {"ID": 10}}
+      ]
+    },
+    {
+      "name": "rowkey should be set when stream has String key and only id specified in insert",
+      "statements": [
+        "CREATE STREAM TEST (ID VARCHAR) WITH (kafka_topic='test_topic', value_format='JSON', key='ID');",
+        "INSERT INTO TEST (ID) VALUES ('10');"
+      ],
+      "inputs": [
+      ],
+      "outputs": [
+        {"topic": "test_topic", "key": "10", "value": {"ID": "10"}}
+      ]
+    },
+    {
+      "name": "rowkey should be set when stream has double key and only id specified in insert",
+      "statements": [
+        "CREATE STREAM TEST (ID DOUBLE) WITH (kafka_topic='test_topic', value_format='JSON', key='ID');",
+        "INSERT INTO TEST (ID) VALUES (1.23);"
+      ],
+      "inputs": [
+      ],
+      "outputs": [
+        {"topic": "test_topic", "key": "1.23", "value": {"ID": 1.23}}
+      ]
+    },
+    {
+      "name": "rowkey should be set when stream has bigint key and only id specified in insert",
+      "statements": [
+        "CREATE STREAM TEST (ID BIGINT) WITH (kafka_topic='test_topic', value_format='JSON', key='ID');",
+        "INSERT INTO TEST (ID) VALUES (10);"
+      ],
+      "inputs": [
+      ],
+      "outputs": [
+        {"topic": "test_topic", "key": "10", "value": {"ID": 10}}
+      ]
+    },
+    {
+      "name": "rowkey should be set when stream has boolean key and only id specified in insert",
+      "statements": [
+        "CREATE STREAM TEST (ID BOOLEAN) WITH (kafka_topic='test_topic', value_format='JSON', key='ID');",
+        "INSERT INTO TEST (ID) VALUES (TRUE);"
+      ],
+      "inputs": [
+      ],
+      "outputs": [
+        {"topic": "test_topic", "key": "true", "value": {"ID": true}}
+      ]
+    },
+    {
+      "name": "id should be set when stream has string key and only rowkey specicified in insert",
```

Likewise

```suggestion
      "name": "key field should be set when stream has string key and only rowkey specified in insert",
```

Same for other tests